runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2219,11 +2219,10 @@ bool TR::CompilationInfo::shouldRetryCompilation(TR_MethodToBeCompiled *entry, T
                break;
 #if defined(J9VM_OPT_JITSERVER)
             case compilationStreamFailure:
-               // This feature TR_RequireJITServer is used when we would like the client to fail when server crashes
-               static char *requireJITServer = feGetEnv("TR_RequireJITServer");
-               if (requireJITServer)
+               // if -XX:+JITServerRequireServer is used, we would like the client to fail when server crashes
+               if (entry->_compInfoPT->getCompilationInfo()->getPersistentInfo()->getRequireJITServer())
                   {
-                  TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE, "In mode TR_RequireJITServer, exit JITClient due to unavailable JITServer.");
+                  TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE, "Option -XX:+JITServerRequireServer is used, exit JITClient due to unavailable JITServer.");
```

Comment:
Does exit code 25 have a special meaning? Isn't better to have a fatal assert here? The message from the assert appears in the test output and we would know why the test failed.

Reply:
I don't know if 25 has any special meaning, it was already present there. I'll just replace it with a fatal assert.
btw, we will always see the verbose log error message because there are no checks for verbose options here.